extensions/kubernetes-client/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientProducer.java

```diff
@@ -0,0 +1,69 @@
+package io.quarkus.kubernetes.client.runtime;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.quarkus.arc.DefaultBean;
+
+@ApplicationScoped
+public class KubernetesClientProducer {
+
+    private KubernetesClientBuildConfig buildConfig;
+
+    @DefaultBean
```

Comment:
Hmmm, this is interesting. I might be able to use it for other things. Nice to see it in action.

Reply:
Yeah, this is what I had in mind when I added the support for `@DefaultBean` :)